src/Common/Admin/Settings_Sidebar.php

```diff
@@ -42,64 +33,77 @@ class Settings_Sidebar extends Section {
 	public function render() {
 		?>
 		<div class="tec-settings-form__sidebar">
-		<?php do_action( 'tec_settings_sidebar_start' ); ?>
-			<div class="tec-settings-form__sidebar-section tec-settings-form__sidebar-header">
-				<?php do_action( 'tec_settings_sidebar_header_start' ); ?>
-				<?php
-				$this->render_header_image();
-				$this->render_title();
-				?>
-				<?php do_action( 'tec_settings_sidebar_header_end' ); ?>
-			</div>
-
-			<?php foreach ( $this->sections as $section ) : ?>
+			<?php do_action( 'tec_settings_sidebar_start', $this ); ?>
+			<?php foreach ( $this->get_sections() as $section ) : ?>
 				<div class="tec-settings-form__sidebar-section">
 					<?php $section->render(); ?>
 				</div>
 			<?php endforeach; ?>
-			<?php do_action( 'tec_settings_sidebar_end' ); ?>
+			<?php do_action( 'tec_settings_sidebar_end', $this ); ?>
 		</div>
 		<?php
 	}
 
 	/**
-	 * Set the header image for the sidebar.
+	 * Alias to prepending a section to the sidebar.
 	 *
 	 * @since 6.1.0
 	 *
-	 * @param Image $image The image to set.
+	 * @param Section $section The section to add.
 	 *
 	 * @return void
 	 */
-	public function set_header_image( Image $image ) {
-		$this->header_image = $image;
+	public function add_section( Section $section ): self {
+		return $this->append_section( $section );
 	}
 
 	/**
-	 * Add a section to the sidebar.
+	 * Add a section to the end of the sidebar array of sections
 	 *
-	 * @since 6.1.0
+	 * @since TBD
 	 *
 	 * @param Section $section The section to add.
 	 *
 	 * @return void
 	 */
-	public function add_section( Section $section ) {
+	public function append_section( Section $section ): self {
 		$this->sections[] = $section;
+
+		return $this;
 	}
 
 	/**
-	 * Render the header image for the sidebar.
+	 * Add a section to the start of the sidebar array of sections
 	 *
-	 * @since 6.1.0
+	 * @since TBD
+	 *
+	 * @param Section $section The section to add.
 	 *
 	 * @return void
 	 */
-	protected function render_header_image() {
-		if ( ! $this->header_image ) {
-			return;
-		}
+	public function prepend_section( Section $section ): self {
+		array_unshift( $this->sections, $section );
 
-		$this->header_image->render();
+		return $this;
 	}
+
+	/**
+	 * Get the sidebar sections.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_sections(): array {
+		/**
+		 * Filter the sidebar sections.
+		 *
+		 * @since TBD
+		 *
+		 * @param Section[]        $sections The sidebar sections.
+		 * @param Settings_Sidebar $sidebar  The sidebar object.
+		 */
+		return apply_filters( 'tec_settings_sidebar_sections', $this->sections, $this );
```

Comment:
We have both, the instance is supplied on a filter elsewhere. I just thought it would be nice to include a filter here.

The only reason I dont like saving back to the `$this->sections` is because it would re-apply the filter to the same `$this->section` if it was called twice back to back.

Reply:
> We have both, the instance is supplied on a filter elsewhere. I just thought it would be nice to include a filter here.

There *is* such a thing as too many filters in too many places where they don't belong. Adding a filter here invites a way to tweak the internals of this class instance in a way that is hard to control. 

If we want to add sections (which is a valid case), then we should use a filter *outside* of this class, and utilize the class methods to provide modifications to the class instance. Otherwise we wind up with spaghetti code that is hard to track down where changes to things happen.